src/lib/util/cdrom.cpp

```diff
@@ -823,6 +823,29 @@ int cdrom_get_track_type(cdrom_file *file, int track)
 }
 
 
+/*-------------------------------------------------
+    cdrom_is_audio_media - heuristically determine
+    type of media based on track types
+-------------------------------------------------*/
+
+int cdrom_get_media_type(cdrom_file *file)
```

Comment:
I finally found CDs that broke the standard format so I added this helper function. PCE-CDs have an audio track as track 1, data as track 2, and then more audio as track 3 and beyond which would fail the check to determine if a CD is audio or mixed/data by checking just track 1.

Reply:
wait until you find the CD-i Ready format ones (The Apprentice etc.) that have the _data_ in the _audio_ pregap of the first track.  the images MAME has for these are incorrect, as cue/bin can't even represent them properly.

Reply:
I didn't realize that it was non-standard. I tried reading through the standard and I still don't feel like I "get" it still but the mixed CD type table seems to at least show the typical(?) layout with data in track 1, so I reverted back to just checking if track 1 is audio or not instead of using the helper function.